sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeExtractors.scala

```diff
@@ -119,13 +120,26 @@ case class GpuGetArrayItem(child: Expression, ordinal: Expression)
     throw new IllegalStateException("This is not supported yet")
 
   override def doColumnar(lhs: GpuColumnVector, ordinalS: GpuScalar): ColumnVector = {
-    // Need to handle negative indexes...
     val ordinal = ordinalS.getValue.asInstanceOf[Int]
-    if (ordinalS.isValid && ordinal >= 0) {
-      lhs.getBase.extractListElement(ordinal)
+    if (ordinalS.isValid) {
+      withResource(lhs.getBase.countElements) { numElementsCV =>
+        withResource(numElementsCV.min) {
+          minScalar =>
+            val minNumElements = minScalar.getInt
+            if ( (ordinal < 0 || minNumElements < ordinal + 1) &&
+              numElementsCV.getRowCount != numElementsCV.getNullCount &&
+              failOnError) {
+              throw new ArrayIndexOutOfBoundsException(
+                s"Invalid index: ${ordinal}, minimum numElements in this ColumnVector: " +
+                  s"$minNumElements")
+            } else {
+              lhs.getBase.extractListElement(ordinal)
+            }
+        }
+      }
     } else {
       GpuColumnVector.columnVectorFromNull(lhs.getRowCount.toInt, dataType)
```

Comment:
How does Spark handle the null case ?

Reply:
Spark will return null, for like null[5].This case is covered by integration_test `test_array_element_at_all_null_ansi_not_fail`